src/fridge.py

```diff
@@ -1,42 +1,69 @@
+from datetime.datetime import now
 import torch
 
 
-def model_saver(model, path):
+def save_model(checkpoint_num, epoch_num, gen, disc, optimizer_gen,
+               optimizer_disc):
     '''
-    NOTE: WIP/TODO
-
-    Takes in a generator or descriminator model and saves the model to an
-    appropriate location for future generation/discrimination
+    Saves model checkpoint.
 
     Parameters
     ----------
-    model: a trained pytorch model
-    path: path to where to save the model
-
-    Returns
-    -------
-    True if it worked false otherwise, possibly something else to be decided
-    later
+    checkpoint_num : int
+        Checkpoint number.
+    epoch_num : int
+        Epoch number.
+    gen : torch.nn.Module
+        Generator object.
+    disc : torch.nn.Module
+        Discriminator object.
+    optimizer_gen : torch.optim.Optimizer
+        Generator optimizer object.
+    optimizer_disc : torch.optim.Optimizer
+        Discriminator optimizer object.
     '''
-    torch.save(model.state_dict(), location)
-    return true
+    now = now().strftime("%m.%d.%H.%M")
+    model = {
+        'epoch': epoch,
+        'gen': gen,
+        'disc': disc,
+        'optimizer_gen': optimizer_gen,
+        'optimizer_disc': optimizer_disc,
+    }
+    state = {
+        'epoch': epoch,
+        'gen_state_dict': gen.state_dict(),
+        'disc_state_dict': disc.state_dict(),
+        'optimizer_gen_state_dict': optimizer_gen.state_dict(),
+        'optimizer_disc_state_dict': optimizer_disc.state_dict(),
+    }
+    torch.save(model, 'checkpoints/models/model_{}_{}.pt'.format(
+        checkpoint_num, now))
+    torch.save(state, 'checkpoints/states/state_{}_{}.pt'.format(
+        checkpoint_num, now))
 
 
-def model_loader(path):
+def load_model(path):
     '''
-    NOTE: WIP/TODO
-
-    Model loader. The point is to be able to load a previously trained model for
-    either continuing training or evaluating the generator model.
+    Loads model checkpoint.
 
     Parameters
     ----------
-    path: path to the model file we wish to open and load
+    path : string
+        Path to the checkpoint file.
 
     Returns
     -------
     pytorch model
-
     '''
-
-    pass
+    if path[:6] == 'model_':
+        checkpoint = torch.load(path)
+        epoch = checkpoint['epoch']
+        gen = checkpoint['gen']
+        disc = checkpoint['disc']
+        optimizer_gen = checkpoint['optimizer_gen']
+        optimizer_disc = checkpoint['optimizer_disc']
+        return epoch, gen, disc, optimizer_gen, optimizer_disc
+    elif path[:6] == 'state_':
+        msg = 'Must instatiate correct class. See https://pytorch.org/docs/stable/notes/serialization.html for more details.'
```

Comment:
If we can't load state, why are we saving it? Is there unwritten load code?

Reply:
Loading state requires you to know which pastry it is, and which optimizers you used. Without this knowledge, I can't load it in.